### PR TITLE
touchscreen: Add EF1E-A1 touch screen driver

### PR DESCRIPTION
**Pull request by phreer**
The touch panel is working fine with this patch.

Test-done:
    Boot Android and evalidate touch functionality.

Tracked-On: OAM-110242

### CODE REVIEW

File: drivers/input/touchscreen/ef1e_tp/tp_driver.c [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+
+
+static int __init tp_driver_init(void)
+{
+	int ret;
+	int i2c_bus_number;
+	struct i2c_adapter *i2c_adap;
+	struct tp_priv *priv = &global_tp;
+
+	if (sizeof (struct tp_report_data) != 65) {
```
Comment (feijiang1) [inline drivers/input/touchscreen/ef1e_tp/tp_driver.c:494]: 65 is a magic number, it is better to define one macro like TP_REPORT_DATA_SIZE

File: drivers/input/touchscreen/ef1e_tp/tp_input.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	if (ret) {
+		pr_err("%s: Failed to initialize MT slots: %d\n", __func__, ret);
+		return ret;
+	}
+	ret = input_register_device(priv->input_dev);
+	if (ret) {
+		pr_err("%s: Failed to register input device\n", __func__);
+		return ret;
+	}
+    return 0;
```
Comment (feijiang1) [inline drivers/input/touchscreen/ef1e_tp/tp_input.c:41]: coding style

File: drivers/input/touchscreen/ef1e_tp/tp.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#ifndef EF1E_TP_H
```
Comment (feijiang1) [inline drivers/input/touchscreen/ef1e_tp/tp.h:1]: need add license header for all files

File: drivers/input/touchscreen/ef1e_tp/Makefile [hunk 4]
```diff
@@ -0,0 +1,18 @@
+ifneq ($(KERNELRELEASE),)
+obj-$(CONFIG_TOUCHSCREEN_EF1E_TP) += ef1e_tp.o
+ef1e_tp-objs += tp_input.o \
+	tp_driver.o
```
Comment (feijiang1) [inline drivers/input/touchscreen/ef1e_tp/Makefile:4]: tp name is too generic, can we change the prefix tp to ef1e?

Comment (phreer) [reply drivers/input/touchscreen/ef1e_tp/Makefile:4]: Sure.

File: drivers/input/touchscreen/ef1e_tp/tp_driver.c [hunk 5]
```diff
@@ -0,0 +1,605 @@
...
+		pr_err("invalid type struct tp_report_data");
+		return -1;
+	}
+	tp_priv_init(priv);
+
+	i2c_bus_number = get_bus_number();
+	pr_info("i2c bus number = %d\n", i2c_bus_number);
+	i2c_adap = i2c_get_adapter(i2c_bus_number);
+	if (!i2c_adap) {
+		pr_debug("Cannot find a valid i2c bus for tp\n");
```
Comment (feijiang1) [inline drivers/input/touchscreen/ef1e_tp/tp_driver.c:504]: should use pr_err?

**Review by feijiang1 — COMMENTED** [review]
also please run scripts/checkpatch.pl to double check the coding style

**Review by feijiang1 — APPROVED** [review]
LGTM

**Comment by feijiang1** [comment]
@xzhan34 @beckwen @wangyiyanga  please also help review

**Review by feijiang1 — APPROVED** [review]
LTGM

**Pull request merged**